internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/utils/UnZipper.java

```diff
@@ -0,0 +1,130 @@
+
+            Enumeration zipFileEntries = zip.entries();
+
+            // Process each entry
+            while (zipFileEntries.hasMoreElements()) {
+                // grab a zip file entry
+                ZipEntry entry = (ZipEntry) zipFileEntries.nextElement();
+                currentEntry = entry.getName();
+                File destFile = new File(newPath, currentEntry);
+                //destFile = new File(newPath, destFile.getName());
+                File destinationParent = destFile.getParentFile();
+
+                // create the parent directory structure if needed
+                boolean isDestcreated = destinationParent.mkdirs();
+
+                if (!entry.isDirectory()) {
+                    BufferedInputStream bufferedInputStream = null;
+                    BufferedOutputStream bufferedOutputStream = null;
+                    InputStream inputStream = null;
+                    try {
+                        inputStream = zip.getInputStream(entry);
+                        bufferedInputStream = new BufferedInputStream(inputStream);
+                        int currentByte;
+                        // establish buffer for writing file
+                        byte data[] = new byte[buffer];
+
+                        // write the current file to disk
+                        FileOutputStream fos = new FileOutputStream(destFile);
+                        bufferedOutputStream = new BufferedOutputStream(fos,
+                                buffer);
+
+                        // read and write until last byte is encountered
+                        while ((currentByte = bufferedInputStream.read(data, 0, buffer)) != -1) {
+                            bufferedOutputStream.write(data, 0, currentByte);
+                        }
+                    } finally {
+                        if (bufferedOutputStream != null) {
+                            bufferedOutputStream.flush();
+                            bufferedOutputStream.close();
+                        }
+                        if (bufferedInputStream != null) {
+                            bufferedInputStream.close();
+                        }
+                        if (inputStream != null) {
+                            inputStream.close();
+                        }
+                    }
+
+                }
+
+                if (currentEntry.endsWith(JSScannerConstants.ZIP_PREFIX)) {
+                    // found a zip file, try to open
+                    extractFolder(destFile.getAbsolutePath());
+                }
+                if (currentEntry.endsWith(JSScannerConstants.JAR_PREFIX)) {
+                    // found a zip file, try to open
+                    extractFolder(destFile.getAbsolutePath());
+                }
+                if (currentEntry.endsWith(JSScannerConstants.WAR_PREFIX)) {
+                    // found a zip file, try to open
+                    extractFolder(destFile.getAbsolutePath());
+
+                }
```

Comment:
Three 'if' statements aren't needed. All of them can be written in a single 'if' using the OR operator.